src/gui/qgsmetadatawidget.cpp

```diff
@@ -107,12 +110,36 @@ QgsMetadataWidget::QgsMetadataWidget( QWidget *parent, QgsMapLayer *layer )
   connect( btnRemoveCategory, &QPushButton::clicked, this, &QgsMetadataWidget::removeSelectedCategory );
 
   fillComboBox();
+  if ( !mLayer )
+  {
+    btnAutoSource->setEnabled( false );
+    btnAutoEncoding->setEnabled( false );
+    btnAutoCrs->setEnabled( false );
+  }
+  setMetadata( mMetadata );
+}
+
+void QgsMetadataWidget::setMetadata( const QgsLayerMetadata &layerMetadata )
+{
+  mMetadata = layerMetadata;
   setPropertiesFromLayer();
 }
 
+QgsLayerMetadata QgsMetadataWidget::getMetadata()
+{
+  if ( saveMetadata( mMetadata ) )
+  {
+   emit metadataChanged();
```

Comment:
I think this signal should be emitted by saveMetadata - it's not expected behavior for a getter to manually trigger a signal emission like this

Reply:
That was the original idea, but because of the const was not possible.
Also saveMetadata  is called elsewhere (when tabValidation is selected), which could cause to many emits.
The emit should be done elsewhere, so that a get is then done - or send the Metadata with the emit.

Reply:
Otherwise I'd just leave off the signal - if it's triggered by a get request anyway then it will only ever be triggered by something calling the getter, and that something could respond in the same way as if the signal was sent.

If the intention here is to detect changes only then I'd rather see an equality operator added to QgsLayerMetadata and the caller of the getter can test for changes using that.

Reply:
Yes, the intention is to detect changes, but I have no idea how an equality operator  works.
So I think, for now,  it best to comment out the checking code to be used as a reference, set saveMetadata back to void and remove the metadataChanged.

Reply:
Sounds good. An equality operator would be great here though - it's basically just moving the code you've written already across to an equality operator within QgsLayerMetadata itself. (Google it - they are a useful c++ technique to be aware of!)